### PR TITLE
Issues/41 validation yaml

### PR DESCRIPTION
**Pull request by AndreyDegtyarev**
Validate yaml file against generated schema converting it to json.
Moved schema creation logic to separate class.

### CODE REVIEW

File: src/Nox.Solution/Resolvers/NoxYamlSerializer.cs [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+        {
+            foreach (var error in results.Errors)
+            {
+                if (results.EvaluationPath.ToString().EndsWith("/$ref")
+                    || error.Key == "type"
+                    )
+                {
+                    continue;
+                }
+                var path = string.IsNullOrEmpty(results.EvaluationPath.ToString()) ? string.Empty : $"Path: {results.EvaluationPath}. ";
```
Comment (rochar) [inline src/Nox.Solution/Resolvers/NoxYamlSerializer.cs:139]: you can avoid an extra string creation:
```suggestion
             string evaluationPath = results.EvaluationPath.ToString();
             var path = string.IsNullOrEmpty(evaluationPath) ? string.Empty : $"Path: {evaluationPath}. ";
```

File: tests/Nox.Solution.Tests/ValidationTests.cs [hunk 2]
```diff
@@ -13,4 +22,30 @@ public void Environment_names_are_unique()
 
         Assert.Equal("The environment name 'test' is duplicated. Environment names must be unique in a solution.", ex.Errors.First().ErrorMessage);
     }
+
+    [Fact]
+    public void Solution_DeserializeWithValidation_ShoouldThrow()
```
Comment (rochar) [inline tests/Nox.Solution.Tests/ValidationTests.cs:27]: a typo on Shouuld...

I would suggest the test name: Deserialize_With_NoxYamlSerializer_ThrowsException, what do y think? Scenario, Method, Expected Result

```suggestion
    public void Deserialize_With_NoxYamlSerializer_ThrowsException()
```

**Review by rochar — CHANGES_REQUESTED** [review]
looks good, two minor suggestions

File: tests/Nox.Solution.Tests/ValidationTests.cs (hunk 2 above)
Comment (AndreyDegtyarev) [reply tests/Nox.Solution.Tests/ValidationTests.cs]: Agree but without extra underline Deserialize_WithNoxYamlSerializer_ThrowsException

**Review by rochar — DISMISSED** [review]

**Review by jan-schutte — APPROVED** [review]

**Review by jan-schutte — COMMENTED** [review]
Some null reference issues in tests

**Review by rochar — APPROVED** [review]

**Review by jan-schutte — APPROVED** [review]

**Pull request merged**